modules/ducktests/tests/ignitetest/tests/snapshot_test.py

```diff
@@ -0,0 +1,134 @@
+        1. Start of ignite cluster.
+        2. Activate cluster.
+        3. Load.
+        4. Idle verify dump_1.
+        5. Snapshot.
+        6. Load.
+        7. Idle verify dump_2.
+        8. Сhecking that dump_1 and dump_2 are different.
+        9. Stop ignite and replace db from snapshot.
+        10. Run ignite cluster.
+        11. Idle verify dump_3.
+        12. Checking the equality of dump_1 and dump_3.
+        """
+        data_storage = DataStorageConfiguration(default=DataRegionConfiguration(persistent=True))
+
+        ignite_config = IgniteConfiguration(
+            version=IgniteVersion(ignite_version),
+            data_storage=data_storage
+        )
+
+        service = IgniteService(self.test_context, ignite_config, num_nodes=self.NUM_NODES - 1)
+        service.start()
+
+        control_utility = ControlUtility(service, self.test_context)
+        control_utility.activate()
+
+        client_config = IgniteConfiguration(
+            client_mode=True,
+            version=IgniteVersion(ignite_version),
+            discovery_spi=from_ignite_cluster(service),
+        )
+
+        streamer = IgniteApplicationService(
+            self.test_context,
+            client_config,
+            java_class_name="org.apache.ignite.internal.ducktest.tests.UuidDataStreamerApplication",
+            params={
+                "cacheName": "test-cache",
+                "iterSize": 512 * 1024
+            }
+        )
+
+        load(streamer)
+
+        node = service.nodes[0]
+
+        control_utility.validate_indexes(check_assert=True)
+        control_utility.idle_verify(check_assert=True)
+        dump_1 = control_utility.idle_verify_dump(node, return_path=True)
+
+        self.logger.info(f'Path to dump_1 on {node.account.externally_routable_ip}={dump_1}')
+
+        control_utility.snapshot_create(self.SNAPSHOT_NAME)
+
+        load(streamer)
+
+        dump_2 = control_utility.idle_verify_dump(node, return_path=True)
+
+        self.logger.info(f'Path to dump_2 on {node.account.externally_routable_ip}={dump_2}')
+
+        diff = node.account.ssh_output(f'diff {dump_1} {dump_2}', allow_fail=True)
+        assert len(diff) != 0
+
+        service.stop()
+
+        service.rename_db(new_db_name='old_db')
+        service.restore_from_snapshot(self.SNAPSHOT_NAME)
+
+        service.restart()
+
+        control_utility.activate()
+
+        control_utility.validate_indexes(check_assert=True)
+        control_utility.idle_verify(check_assert=True)
+        dump_3 = control_utility.idle_verify_dump(node, return_path=True)
+
+        self.logger.info(f'Path to dump_3 on {node.account.externally_routable_ip}={dump_3}')
+
+        diff = node.account.ssh_output(f'diff {dump_1} {dump_3}', allow_fail=True)
+        assert len(diff) == 0, diff
+
+
+def load(service_load: IgniteApplicationService, duration: int = 300):
+    """
+    Load.
+    """
+    service_load.start()
+
+    service_load.await_stopped(duration)
```

Comment:
why not just .run()?

Reply:
.run() will fail in .stop() because there is a timeout of 10 seconds.

Reply:
fixed